processing/src/main/java/org/apache/druid/query/aggregation/post/JavaScriptPostAggregator.java

```diff
@@ -136,19 +143,20 @@ public Object compute(Map<String, Object> combinedAggregators)
    * {@link #compute} can be called by multiple threads, so this function should be thread-safe to avoid extra
    * script compilation.
    */
+  @EnsuresNonNull("fn")
   private void checkAndCompileScript()
   {
-    if (fn == null) {
-      // JavaScript configuration should be checked when it's actually used because someone might still want Druid
-      // nodes to be able to deserialize JavaScript-based objects even though JavaScript is disabled.
-      Preconditions.checkState(config.isEnabled(), "JavaScript is disabled");
-
-      // Synchronizing here can degrade the performance significantly because this method is called per input row.
-      // However, early compilation of JavaScript functions can occur some memory issues due to unnecessary compilation
-      // involving Java class generation each time, and thus this will be better.
+    // JavaScript configuration should be checked when it's actually used because someone might still want Druid
+    // nodes to be able to deserialize JavaScript-based objects even though JavaScript is disabled.
+    Preconditions.checkState(config.isEnabled(), "JavaScript is disabled");
+
+    Function syncedFn = fn;
+    if (syncedFn == null) {
```

Comment:
@leventov Seems that here is always null at new case. Do I miss anything?

Reply:
I don't understand you message.

Reply:
Seems that   first check for   `syncedFn`

```java
if (syncedFn == null)
```
is unnecessary. 

On the other hand, I've just sent another commit to address your comments.

Reply:
@leventov is everything OK for my PR?